os/net/lwip/src/include/lwip/lwipopts.h

```diff
@@ -652,15 +652,27 @@
 #define LWIP_RANDOMIZE_INITIAL_LOCAL_PORTS      CONFIG_NET_RANDOMIZE_INITIAL_LOCAL_PORTS
 #endif
 
-#ifdef CONFIG_NFILE_DESCRIPTORS
-#define LWIP_SOCKET_OFFSET CONFIG_NFILE_DESCRIPTORS
+#if !defined(CONFIG_NSOCKET_DESCRIPTORS) || (CONFIG_NSOCKET_DESCRIPTORS == 0)
+#error "CONFIG_NSOCKET_DESCRIPTORS is undefined or 0"
 #endif
 
-#ifdef CONFIG_NSOCKET_DESCRIPTORS
-#define MEMP_NUM_NETCONN CONFIG_NSOCKET_DESCRIPTORS
-#define MEMP_NUM_RAW_PCB CONFIG_NSOCKET_DESCRIPTORS
+#ifndef CONFIG_NFILE_DESCRIPTORS
+#define CONFIG_NFILE_DESCRIPTORS 0
```

Comment:
I think this checking is unnecessary. we should check dependencies in Kconfig

Reply:
Then I'll remove this code.